core/include/gnuradio-4.0/Block.hpp

```diff
@@ -1212,20 +1213,22 @@ class Block : public lifecycle::StateMachine<Derived>, //
     auto
     getNextTagAndEosPosition() {
         struct {
-            bool hasTag = false;
-            std::size_t nextTag = std::numeric_limits<std::size_t>::max();
+            bool        hasTag     = false;
+            std::size_t nextTag    = std::numeric_limits<std::size_t>::max();
             std::size_t nextEosTag = std::numeric_limits<std::size_t>::max();
-            bool asyncEoS = false;
+            bool        asyncEoS   = false;
         } result;
-        auto        adjustForInputPort = [&result]<PortLike Port>(Port &port) {
+
+        auto adjustForInputPort = [&result]<PortLike Port>(Port &port) {
             if (port.isConnected()) {
                 if constexpr (std::remove_cvref_t<Port>::kIsSynch) {
                     // get the tag after the one at position 0 that will be evaluated for this chunk.
                     // nextTag limits the size of the chunk except if this would violate port constraints
                     result.nextTag                        = std::min(result.nextTag, nSamplesUntilNextTag(port, 1).value_or(std::numeric_limits<std::size_t>::max()));
                     result.nextEosTag                     = std::min(result.nextEosTag, samples_to_eos_tag(port).value_or(std::numeric_limits<std::size_t>::max()));
-                    const gr::ConsumableSpan auto tagData = port.tagReader().get(port.tagReader().available());
-                    result.hasTag =  result.hasTag || (!tagData.empty() && tagData[0].index == port.streamReader().position() && !tagData[0].map.empty());
+                    const gr::ConsumableSpan auto tagData = port.tagReader().get();
+                    result.hasTag                         = result.hasTag || (!tagData.empty() && tagData[0].index == port.streamReader().position() && !tagData[0].map.empty());
+                    std::ignore                           = tagData.consume(0UZ);
```

Comment:
Couldn't this `consume(0UZ)` be avoided by setting the appropriate policy on the calling `get<SpanReleasePolicy::ProcessNone>()'?

Reply:
This line is optional as long as default policy is `ProcessNone`. I remove it.